src/idpi/grib_decoder.py

```diff
@@ -32,8 +31,6 @@
     "isobaricInPa": ("pressure", 0.0),
     "surface": ("surface", 0.0),
 }
-_ifs_allowed = True
-_cosmo_allowed = True
```

Comment:
I don't think this removes the need to exclusively read cosmo or ifs data. The mapping is no longer applied but the actual issue was in the stale eccodes definitions cache that can not be cleared.

Reply:
right, the config does not remove the need for this. However, this protection has also its pitfalls in my opinion. One needs to set those variables properly, which is error prone. 
There are several places with 
` idpi.config.get("data_scope", "cosmo") == "cosmo"`
that required setting the variables. 
If someones adds a new function that loads additional fields or require entering the scope of cosmo, ifs data, they should not forget that this variables must be set. 
Trying to avoid that, before we had wrapper functions duplicated for ifs and cosmo that are just there so that these variables are not forgotten. Which pollutes considerably the API. 
I think is a trade off, but provided that mixing IFS data and COSMO data in our processing is probably not a use case at this point in time, I thought the solution does not pay off. 
At some point, some projects might need to combine the processing of various data sources, but then I would argue that we should invest in fixing the issue with eccodes, rather than trying to protect our codebase. I do not see any fundamental reason why we could not clear the cache of eccodes, it is probably just a bug. 
Thoughts on this?